File: C/conditional2.c
/*
 * Demonstrate the switch statement
 *
 * Compile with: gcc -o conditional2 conditional2.c
 */

#include <stdio.h>
#include <stdbool.h>

int main(void){
	char question[] = "Enter a number between 1 and 4: ";
	
	while(true){
        /* Query for a number between 1 and 4 */
		printf(question);
		int guess;
		scanf("%d", &guess);
		bool inrange = true;
		
        /* Evaluate the response */
		switch(guess){
			case 1:
				printf("You entered the number One\n");
                break;
            case 2:
				printf("You entered the number Two\n");
                break;
            case 3:
                printf("You entered the number Three\n");
                break;
            case 4:
                printf("You entered the number Four\n");
                break;
            default:
                /* Error condition */
                printf("Please try again!\n");
                inrange = false;
                break;
		}
        if(inrange){
            break;
        }
    }
    return(0);
}

File: Java/Conditional2.java
// Demonstrate switch statement

import java.util.Scanner;

public class Conditional2 {
    public static void main(String [] args) {

        // Method to accept input from standard input
        Scanner in = new Scanner(System.in);
    
        String question = "Enter a number between 1 and 4: ";
    
        while(true){
            // Ask for a number
            System.out.printf(question);
            int guess = in.nextInt();
            boolean inrange = true;
    
            // Identify the number
            switch(guess){
                case 1:
                    System.out.println("You entered the number One");
                    break;
                case 2:
                    System.out.println("You entered the number Two");
                    break;
                case 3:
                    System.out.println("You entered the number Three");
                    break;
                case 4:
                    System.out.println("You entered the number Four");
                    break;
                default:
                    System.out.println("Please try again!");
                    inrange = false;
                    break;
                    
            }

            // Check if the user can follow directions
            if(inrange){
                break;
            }
        }
    }
}

